src/printer.js

```diff
@@ -726,8 +726,27 @@ function printPathNoParens(path, options, print) {
       }
 
       return concat(parts)
-    case 'ExpressionStatement':
-      return path.call(print, 'expression')
+    case 'ExpressionStatement': {
+      const printed = path.call(print, 'expression')
+      if (!expressionStatementHasInlineLeadingComment(n, options)) {
+        return printed
+      }
+
+      n.comments
```

Comment:
This is kind of ugly. Basically I'm manually simulating eg Prettier's `printLeadingComment()`

Feels like the deeper issue here is that you either call Prettier's `printComments()` or do it all yourself (which presumably you're not really supposed to do?), and `printComments()` isn't flexible enough in this case